cmd/nerdctl/container_cp_linux_test.go

```diff
@@ -25,396 +25,933 @@ import (
 	"testing"
 
 	"gotest.tools/v3/assert"
+	"gotest.tools/v3/icmd"
 
+	"github.com/containerd/nerdctl/v2/pkg/containerutil"
 	"github.com/containerd/nerdctl/v2/pkg/rootlessutil"
 	"github.com/containerd/nerdctl/v2/pkg/testutil"
 )
 
+// For the test matrix, see https://docs.docker.com/engine/reference/commandline/cp/
+// Obviously, none of this is fully windows ready - obviously `nerdctl cp` itself is not either, so, ok for now.
+const (
+	// Use this to poke the testing rig for improper path handling
+	// TODO: fuzz this more seriously
+	// FIXME: the following will break the test (anything that will evaluate on the shell, obviously):
+	// - `
+	// - $a, ${a}, etc
+	complexify = "" //  = "-~a0-_.(){}[]*#! \"'∞"
```

Comment:
Useful to stress test rig when modifying it. Deactivated by default as it makes things much harder to read.